src/CastleAdventure.java

```diff
@@ -3,29 +3,170 @@
+
+        System.out.println("\nIn next room you see the dragon! And he starts to attack you!");
+
+        int[] dragonAttacks = {8, 16, 24, 32, 40};
+
+        System.out.print("Dragon attacks you with such power: ");
+        for (int attack : dragonAttacks) {
+            System.out.print(attack + " ");
+        }
+        System.out.println("\n");
+
+        boolean dragonAnswer = false;
+    do {
+        System.out.println("What we can say about these attacks?");
+        System.out.println("1 - Attacks increase by 10");
+        System.out.println("2 - All attacks are multiples of three");
+        System.out.println("3 - All attacks are even");
+        System.out.println("4 - There are both even and odd attacks");
+        System.out.println("5 - All attacks are the same");
+
+        System.out.print("Select correct option (1-5): \n");
+        int choice3 = denScanner.nextInt();
+
+        if (choice3 == 3) {
+            System.out.println("Correct! All attacks are even! Shield activated!");
+            System.out.println("Doors to the net room are opened");
+            dragonAnswer = true;
         } else {
-            System.out.println("Невірний код. Спробуйте ще.");
+            System.out.println("Incorrect! Dragon attacks you again!");
+            System.out.println("Get your strength back and try again...");
         }
+    } while (!dragonAnswer);
+
+
+        System.out.println("\nFinal room: Find the magic key!");
+
+/* our prev arrays:
+        int[] secretCode = {5, 7, 13, 49, 125};
+        int[] numbersBoard = {3, 6, 9, 12, 15, 18, 21, 24, 27, 30};
+        int[] dragonAttacks = {8, 16, 24, 32, 40};
+*/
+        boolean isKeyFound = false;
+
+        do {
+            System.out.print("Type your favorite number: ");
+            int favoriteNumber = denScanner.nextInt();
+
+            for (int userFavNum : secretCode) {
+                if (userFavNum == favoriteNumber) {
+                    isKeyFound = true;
+                    break;
+                }
+            }
+            if (!isKeyFound) {
+                for (int userFavNum : numbersBoard) {
+                    if (userFavNum == favoriteNumber) {
+                        isKeyFound = true;
+                        break;
+                    }
+                }
+            }
+            if (!isKeyFound) {
+                for (int userFavNum : dragonAttacks) {
+                    if (userFavNum == favoriteNumber) {
+                        isKeyFound = true;
+                        break;
+                    }
+                }
+            }
+
+            if (!isKeyFound) {
+                System.out.println("Need more magic... Try again!\n");
+            }
+
+        } while (!isKeyFound);
 
-        // Інші частини пригоди додайте тут!
+        System.out.println("You found the magic key! You are true mag! See you soon in next adventures!");
+        denScanner.close();
+        System.exit(0);
```

Comment:
This one is not really needed to be specified explicitly as compiler reached end of the program anyway

Reply:
Usually it's always better to let your program finish naturally, unless you really need to throw specifix exit code or interrupt it